File: Segundo Trimestre/5- Ejercicios Colecciones y Diccionarios/Python/Ejercicio2.py
"""
Ejercicio 2
Realiza un programa que introduzca valores aleatorios (entre 0 y 100) en un
ArrayList y que luego calcule la suma, la media, el máximo y el mínimo de
esos números. El tamaño de la lista también será aleatorio y podrá oscilar
entre 10 y 20 elementos ambos inclusive.

@author: RafaLpeC
"""

from random import randint

suma = 0
minimo = 100
maximo = 0
tamano = randint(10, 20) 

numeros = [randint(0, 100) for _ in range(tamano)]

print("Lista generada:" , numeros)

for i in numeros:
    suma += i

    if (i < minimo):
        minimo = i

    if (i > maximo):
        maximo = i

print("La suma es igual a" , suma);
print("La media es igual a" , suma / tamano);
print("El número más grande es" , maximo,"y el más pequeño es",maximo);

File: Segundo Trimestre/5- Ejercicios Colecciones y Diccionarios/Java/Ejercicio2.java
import java.util.ArrayList;

public class Ejercicio2 {
  
  public static int suma(ArrayList<Integer>numeros) {
    int sum = 0;
    for(int i = 0; i < numeros.size();i++) {
      sum += numeros.get(i);
    }
    return sum;
  }
  
  public static int media(ArrayList<Integer>numeros) {
    return suma(numeros)/numeros.size();
  }
  
  public static int grande(ArrayList<Integer>numeros) {
    int mayor = numeros.get(0);
    for(int i = 1; i < numeros.size();i++) {
      if(numeros.get(i)>mayor) {
        mayor = numeros.get(i);
      }
    }
    return mayor;
  }
  
  public static int pequenio(ArrayList<Integer>numeros) {
    int menor = numeros.get(0);
    for(int i = 1; i < numeros.size();i++) {
      if(numeros.get(i)<menor) {
        menor = numeros.get(i);
      }
    }
    return menor;
  }
  
  public static void main(String[] args) {
    ArrayList<Integer> numeros = new ArrayList<Integer>();
    
    int tamaniolista = 10 + (int)(Math.random() * ((20 - 10) + 1));
    
    for(int i = 0; i<tamaniolista;i++) {
      numeros.add((int)(Math.random() * 100 + 1));
      System.out.println(numeros.get(i));
    }
    
    System.out.println("La suma es igual a " + suma(numeros));
    System.out.println("La media es igual a " + media(numeros));
    System.out.println("El número más grande es " + grande(numeros) + " y el más pequeño es " + pequenio(numeros));
  }
}
